docs/configuration/alerting_rules.md

```diff
@@ -32,16 +32,14 @@ groups:
 ```
 
 The optional `for` clause causes Prometheus to wait for a certain duration
-between first encountering a new expression output vector element (like an
-instance with a high HTTP error rate) and counting an alert as firing for this
-element. Elements that are active, but not firing yet, are in pending state.
+between first encountering a new expression output vector element and counting an alert as firing for this element. This helps ensure an alert is valid. In this case, Prometheus will ensure the expression output vector elements are consistent for 10 minutes before firing the alert. Elements that are active, but not firing yet, are in the pending state.
```

Comment:
This isn't 100% accurate, as this is checked at each evaluation rather than being a delayed check that happens independently of the eval cycle.

> Prometheus will ensure the expression output vector elements are consistent for 10 minutes before firing the alert.

This is not correct, the check is done per time series not across the alerting expression. I'd leave this out.

> This helps ensure an alert is valid.

This doesn't fit in reference docs, especially as "valid" could be misconstrued to be a Prometheus notion.